src/net_processing.cpp

```diff
@@ -1754,7 +1757,10 @@ void static ProcessGetData(CNode& pfrom, const CChainParams& chainparams, CConnm
 {
     AssertLockNotHeld(cs_main);
 
-    std::deque<CInv>::iterator it = pfrom.vRecvGetData.begin();
+    PeerRef peer = GetPeerRef(pfrom.GetId());
+    if (peer == nullptr) return;
+
+    std::deque<CInv>::iterator it = peer->vRecvGetData.begin();
```

Comment:
2d9f2fca43aadcdda4d644cddab36dca88b40b97, nit:
```suggestion
    auto it = peer->vRecvGetData.begin();
```

Reply:
We don't have guidance about `auto` usage in our style guide. I personally think `auto` should only be used if the type is immediately obvious from the surrounding code, and it's saving redundant and verbose keystrokes.

Reply:
> We don't have guidance about `auto` usage in our style guide.

Yes. And I think we should have it.

> I personally think `auto` should only be used if the type is immediately obvious from the surrounding code, and it's saving redundant and verbose keystrokes.

https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/

Reply:
Thanks @hebasto. I hadn't seen that particular blog post before, but I am familiar with the arguments for and against `auto`. Scott Meyers also dedicates a chapter to it in Exceptional Modern C++.

Here, I think it's useful to explicitly name the type, since `CInv` is part of the interface. Sure, we could switch out `std::deque` for another container at some point, but to know what `it->IsGenTxMsg()` is doing a few lines further down, you need to know that `it` is an iterator over some container of `CInv`s. I'd much rather the code told me that explicitly than having to go find out what `m_getdata_requests` is from somewhere else.

Reply:
ok.

Reply:
That is a very interesting post! I'm not sure I entirely agree with his reasoning against the readability argument, but perhaps we could debate that somewhere else :) I think this could go either way, and I prefer having the type explicit.